src/components/VideoPlayer.vue

```diff
@@ -1,105 +1,106 @@
-                        self.currentTime = player.getCurrentTime();
-                    
-                      if (self.loopStart && self.loopEnd && self.currentTime > self.loopEndSeconds) {
-                          console.log("here");
-                          player.seekTo(self.loopStartSeconds, true);
-                      }
-                  }, 250)
-              } else if (event.data != YT.PlayerState.PLAYING) {
-                  console.log("ending loop");
-                  clearInterval(this.checkInterval);
-              }
-          },
-          playVideo() {
-                player.seekTo(this.loopStartSeconds, true);
-                player.playVideo();
-          },
-          stopVideo() {
-              player.stopVideo();
-          }
+        loopStartSeconds() {
+            return this.loopStart * 60;  
         },
-        mounted() {
-            function onYouTubeIframeAPIReady() {
-                player = new YT.Player("player", {
-                    height: "390",
-                    width: "640",
-                    videoId: this.videoId,
-                    events: {
-                        onReady: this.onPlayerReady,
-                        onStateChange: this.onPlayerStateChange
+        loopEndSeconds() {
+            return this.loopEnd * 60;
+        }
+    },
+    watch: {
+        loopStart() {
+            player.seekTo(this.loopStartSeconds, true);
+        }
+    },
+    methods: {
+        onPlayerReady(event) {
+            //event.target.playVideo();
+            this.duration = player.getDuration()/60;
+        },
+        onPlayerStateChange(event) {
+            if (event.data == YT.PlayerState.PLAYING && !done) {
+                var self = this;
+                console.log("starting loop");
+                
+                this.checkInterval = setInterval(function() {
+                    if (player.getCurrentTime)  
+                    self.currentTime = player.getCurrentTime();
+                
+                    if (self.loopStart && self.loopEnd && self.currentTime > self.loopEndSeconds) {
+                        console.log("here");
+                        player.seekTo(self.loopStartSeconds, true);
                     }
-                })
-            };
+                }, 250)
+            } else if (event.data != YT.PlayerState.PLAYING) {
+                console.log("ending loop");
+                clearInterval(this.checkInterval);
+            }
+        },
+        playVideo() {
+            player.seekTo(this.loopStartSeconds, true);
+            player.playVideo();
+        },
+        stopVideo() {
+            player.stopVideo();
+        },
+        getPlayer() {
+            return player;
         }
+    },
+    mounted() {
+        player = YouTubePlayer(this.playerId);
+        player.loadVideoById(this.videoId);
+        player.on("ready", ev => this.onPlayerReady(ev));
+        player.on("stateChange", ev => this.onPlayerStateChange(ev));
```

Comment:
The setup for this package is a different:
https://github.com/gajus/youtube-player/blob/master/src/YouTubePlayer.js